internal/controller/sleepinfo/jsonpatch/genericresources.go

```diff
@@ -109,11 +123,31 @@ func getFieldToExclude(excludeRef []v1alpha1.ExcludeRef, target v1alpha1.PatchTa
 	return fieldsSelector
 }
 
+func getFieldToInclude(includeRef []v1alpha1.IncludeRef, target v1alpha1.PatchTarget) []string {
+	var names []string
+	for _, include := range includeRef {
+		if matchPatchTargetAndIncludeRef(target, include) && include.Name != "" {
+			names = append(names, include.Name)
+		}
+	}
+
+	fieldsSelector := []string{}
+	for _, name := range names {
+		fieldsSelector = append(fieldsSelector, fmt.Sprintf("metadata.name==%s", name))
+	}
+	return fieldsSelector
+}
+
 // TODO: check when add support to versions
 func matchPatchTargetAndExcludeRef(target v1alpha1.PatchTarget, excludeRef v1alpha1.ExcludeRef) bool {
 	return strings.HasPrefix(excludeRef.APIVersion, fmt.Sprintf("%s/", target.Group)) && excludeRef.Kind == target.Kind
 }
 
+// TODO: check when add support to versions
+func matchPatchTargetAndIncludeRef(target v1alpha1.PatchTarget, includeRef v1alpha1.IncludeRef) bool {
+	return strings.HasPrefix(includeRef.APIVersion, fmt.Sprintf("%s/", target.Group)) && includeRef.Kind == target.Kind
+}
+
```

Comment:
Instead of declare this function, that seems to me equal to the `matchPatchTargetAndExcludeRef`, why not rename that function to `matchPatchTargetAndRef` and reuse it?

Reply:
AFAICT, this is possible by 2 ways:
- Implementing an interface by both ExcludeRef and IncludeRef types so matchPatchTargetAndRef can accept that as second parameter
- Getting rid of IncludeRef type in api/v1alpha1 since it has the same parameters usage and using the same type everywhere

Which one do you mean? @davidebianchi

Reply:
I meant the second one. So, create a unique reference type which can be reused to specify both include and exclude references.

Reply:
Will do after making sure e2e tests pass with the inclusion feature

Reply:
This is now implemented